sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +354,144 @@ export class AssessmentSequentialProcessingService {
       options?.alternativeReferenceDate,
     );
   }
+
+  /**
+   * Get SFAS application part time awards data from either the SFAS application table or
+   * the SFAS part time applications table depending on the SFAS application query type.
+   * @param assessmentId Current assessment being processed.
+   * @param sfasApplicationType SFAS application query type (sfasApplication or sfasPTApplication).
+   * @returns SFAS application part time awards data.
+   */
+  async getProgramYearLegacyAwardsTotals(
```

Comment:
I would not recommend to have one generic method to get part-time and full-time legacy awards.